### PR TITLE
feat: Include updated images in commit message

### LINKED ISSUE #48: Ability to customize commit message
I think it would be nice to be able to customize the commit message. Specifically, I am wanting to make it less generic and put the image that is being updated, so maybe it could be in the decoder script.

### PR DESCRIPTION
**Pull request by tsmalls93**
Fixes #48

### CODE REVIEW

**Comment by bluebrown** [comment]
Dont worry about the ci failure. I put a failing test in main, on purpose. I will fix that one now, then you could rebase main to get this CI green.

**Comment by bluebrown** [comment]
I have fixed the test https://github.com/bluebrown/kobold/commit/a92a2d2fce1c2836d5fc74bd391eb18e4bfc8595

**Comment by tsmalls93** [comment]
> Hi, thank you for the effort! I am a little bit confused about this design. Like moving it into the decoder.
> 
> What is the benefit of this vs ranging over the actual changes reported by the krm filter?

So, the solution you had in mind was to construct the commit message from the changes array that is returned from the `Pipeline` function? That probably would have been easier if I had thought of that. I suppose the only benefit to my solution is it gives more control to the user, but either solution would work for my use case.

**Comment by tsmalls93** [comment]
Oh, I see. The image(s) being sent to the decoder won't necessarily result in a change.

**Comment by bluebrown** [comment]
> Oh, I see. The image(s) being sent to the decoder won't necessarily result in a change.

Yes, that is one thing. And another thing is that multiple events can could run as batch. So it may change seral image refs in your repo at once.  Depending on the debounce timer and so on.

I did experiment with tracking changes in the past, doing file diffs and so on. But at the end a _semantic diff_ is probably the best, on order to avoid false positives due to things like white space and indentation changes.

So I found it most solid to have the filter report track and report the changes it makes.

File: task/handler.go [hunk 1]
```diff
@@ -43,8 +43,10 @@ func KoboldHandler(ctx context.Context, cache string, g model.TaskGroup, runner
 		g.DestBranch.String = g.RepoUri.Ref
 		g.DestBranch.Valid = true
 	}
-
-	msg = "chore(kobold): update image refs"
+	msg, err = GetCommitMessage(changes)
```
Comment (bluebrown) [inline task/handler.go:46]: I think we could make changes in a slice of structs with structured data so we don't have to use regex.

Comment (bluebrown) [reply task/handler.go:46]: I guess here the type needs to be changed to some exported struct type: https://github.com/bluebrown/kobold/blob/a92a2d2fce1c2836d5fc74bd391eb18e4bfc8595/krm/filter.go#L53

**Comment by bluebrown** [comment]
you can run the checks locally https://github.com/bluebrown/kobold/blob/a92a2d2fce1c2836d5fc74bd391eb18e4bfc8595/Makefile#L40. golangci has also something like a --fix flag, Iu think.

**Comment by tsmalls93** [comment]
> you can run the checks locally
> 
> https://github.com/bluebrown/kobold/blob/a92a2d2fce1c2836d5fc74bd391eb18e4bfc8595/Makefile#L40
> 
> . golangci has also something like a --fix flag, Iu think.

Yeah, I'm getting different errors and the linting seems to give up after them. It is not liking the `copyDir` file seemingly because I am on an arm64 Mac and not x86_64 Linux. I think I got the import issues fixed though.

**Comment by tsmalls93** [comment]
Maybe I can make an Apple Silicon specific Makefile in a future PR.

**Comment by bluebrown** [comment]
> > you can run the checks locally
> > https://github.com/bluebrown/kobold/blob/a92a2d2fce1c2836d5fc74bd391eb18e4bfc8595/Makefile#L40
> > 
> > . golangci has also something like a --fix flag, Iu think.
> 
> Yeah, I'm getting different errors and the linting seems to give up after them. It is not liking the `copyDir` file seemingly because I am on an arm64 Mac and not x86_64 Linux. I think I got the import issues fixed though.

Go has build tags. Some of them are architecture specific. As a shortcut, if you suffix a file with an architecutre, it will be only build on that arch, like if you would have used a build tag.

You could create a file called `copy_darwin.go`  and put some mac specific copy command there. The reason its guarded is because its using exec to run an external command.

**Comment by tsmalls93** [comment]
Gotcha. I ended up renaming it to copy.go and adding `//go:build linux || darwin` to the top of the file, since it would be the same code. That made the error go away. I didn't commit it since my build issues are out of scope for this PR, but I'll do some work to get it to be compatible with arm64 Darwin.

**Comment by bluebrown** [comment]
Hi, thanks. Do you think its good now ? Should we maybe add a few tests?

**Comment by tsmalls93** [comment]
Yeah, I think it is good now. I just added a couple tests.

**Comment by bluebrown** [comment]
hi @tsmalls93 , thanks. I build the latest main as image: docker.io/bluebrown/kobold:aab019c

Its not released yet but you can use that one already, if you want.

**Pull request merged**